core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -444,32 +444,21 @@ object ConfigCommand extends Config {
   }
 
   private[admin] def parseEntity(opts: ConfigCommandOptions): ConfigEntity = {
-    val entityTypes = opts.options.valuesOf(opts.entityType).asScala
+    val entityTypes = opts.entityTypes
+    val entityNames = opts.entityNames
     if (entityTypes.head == ConfigType.User || entityTypes.head == ConfigType.Client)
-      parseQuotaEntity(opts)
+      parseQuotaEntity(opts, entityTypes, entityNames)
     else {
       // Exactly one entity type and at-most one entity name expected for other entities
-      val name = if (opts.options.has(opts.entityName))
-        Some(opts.options.valueOf(opts.entityName))
-      else if (entityTypes.head == ConfigType.Broker && opts.options.has(opts.entityDefault))
-        Some(ConfigEntityName.Default)
-      else
-        None
+      val name = entityNames.headOption match {
+        case Some("") => Some(ConfigEntityName.Default)
```

Comment:
I know this is an existing problem in the code, but I really think we should avoid overloading the meaning of the empty string to mean "the default."  It's not very clear, and furthermore, we made the unfortunate decision to allow empty names in some cases.  For example empty group names are valid. :(  Let's use a `List[Option[String]]` here to make it clear.

Reply:
Agreed, this would be much cleaner. The only concern I have is that this may break compatibility, e.g. --entity-type users --entity-name "" (double quotes) would resolve to the default user today, whereas with this change it'd resolve to the empty user name. If we continue to pass it along, then the formed ZK config path may be invalid/unexpected.

I would consider leaving it as-is for ConfigCommand, and then the new API can be more intelligent about this sort of thing. Thoughts?

Reply:
Ping - any thoughts/comments? BTW, you will need to integrate the PR since I don't have permissions.

Reply:
I guess we can kick the can down the road with regard to empty string handling.  We will have to tackle it when designing the client side quotas API, though (probably by forbidding empty strings for now...)